### PR TITLE
(maint) Fix #pick_engine method

### PR DESCRIPTION
**Pull request by e-gris**
My first try at isolating the pick_engine method was buggy. Cas provided
us a better one.

Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

**Review by donoghuc — APPROVED** [review]
This looks better to me. I dont have a local repro case though i was using CI...

**Comment by e-gris** [comment]
I'm away for the next couple of days, so I'll run some tests and release next week if we don't find any issues.

File: lib/vanagon/driver.rb [hunk 1]
```diff
@@ -56,26 +56,22 @@ def filter_out_components(only_build)
     end
 
     def pick_engine(options) # rubocop:disable Metrics/PerceivedComplexity
```
Comment (donoghuc) [inline lib/vanagon/driver.rb:58]: Hopefully this appeases rubocop and we dont have to disable it... i think generally having a ton of return statements in a ruby method will rasie some eyebrowns.

Comment (e-gris) [reply lib/vanagon/driver.rb:58]: I tried without and rubocop complained. 

Typically, I like guardian clauses over if/elses but I agree my previous implementation was too confusing.

Comment (donoghuc) [reply lib/vanagon/driver.rb:58]: ah, i had that disabled locally in a `.rubocop.yml`. sorry.

**Review by ciprianbadescu — APPROVED** [review]

**Review by underscorgan — APPROVED** [review]

**Pull request merged**